### ISSUE TITLE
Add option to log in json format

### ISSUE THREAD

**Issue by errm** [containous/traefik#780]
It is much nicer to log access and error logs as json if you want them to be ingested into some logging system e.g. ELK, Stackdriver, Papertrail whatever. 

This allows us to add extra fields to logs without worrying that we might be breaking peoples log ingestion.

It also would be good to add an option to log access and error logs to stdout and stderr, since some systems are setup to ingest that without further config.  e.g. kubernetes

**Comment by SirCAS** [comment]
I have started on a proposal for how we can implement a way to print the output from the access logs as json.  I still trying to get my head around golang and the libraries Traefik uses. I'm not sure if I'm heading in the right direction with the current implementation.

The main concerns I have about the current implementation is:
- The Traefik daemon must be restarted when the config value for LogFormat is set. Is this acceptable?
- I have made two const string which contains the possible configuration values for LogFormat. The best solution would be to group the values by making a LogFormat type. My attempts on that have failed so far. Flaeg keeps complaining about a missing parser for the type and I had no luck implementing one (are custom parser only ment for collections?). 

Would anyone mind having a look at my progress so far (I have not looked at the tests or documentation yet)? Any suggestions or feedback is much appreciated (a link is below).

Also, should it be possible to configure different LogFormats for the access log file and the Traefik error log?

https://github.com/SirCAS/traefik/commit/419dc0491f03e1625369313b81a6ab74dc820d40
